### DISCUSSION TITLE
exceptions loading saved config files, 3.0.2

### DISCUSSION THREAD

**Discussion by oldsparks** [artisan-roaster-scope/artisan#1692] [Q&A]
I'm setting up a new Windows 11 PC. I installed the latest release, 3.0.2 on the new machine. The old laptop, also Windows 11 had 3.0.0 installed. I was able to load older config files that were pre 3.0 but the config files I have from the 3.0 install throw exception errors at various lines. I'm not certain of which version the older config file were from but likely a 2.10.n version. 

I have 3 config files saved from the 3.0.0 and none of them will load on this new PC with 3.0.2. 

I plan on installing the 3.0.2 on the old PC and then see if I can load and save the config files on that machine and then hopefully they'll open on the new machine as well. Alternatively I'll have to try rolling back to 3.0.0 on the new machine.

**Comment by MAKOMO** [comment]
Installing 3.0.2 on the old PC might not work as this most likely does not depend on the PC version you are using.

Do you mind sharing those setup files which throw exceptions and show us the exception errors that are thrown such that we can investigate?

**Reply by oldsparks** [reply]
I'm not sure the best way to share the config files so I've just saved the as a .txt file in order to get them uploaded. Change the suffix back to .aset if needed. 

When loading Artisan-Yoshan_Testing I get an exception in line 16889.
File ...6kPa throws exception in line 16586 and18037.
File ...5kPa throws exception in line 16885. 

[artisan-Yoshan_5kPa.txt](https://github.com/user-attachments/files/16974461/artisan-Yoshan_5kPa.txt)
[Artisan-Yoshan_6kPa.txt](https://github.com/user-attachments/files/16974463/Artisan-Yoshan_6kPa.txt)
[Artisan-Yoshan_Testing.txt](https://github.com/user-attachments/files/16974465/Artisan-Yoshan_Testing.txt)

**Reply by MAKOMO** [reply]
Perfect! Thanks for your detailed and very productive report. You did the upload correctly. GitHub does only allow files with extension .txt to be uploaded via drag-and-drop thus .aset files (which are actually just text files) have to be renamed for upload (and after download).

I can duplicate the issues and will fix them today. I will report here once done.

**Reply by MAKOMO** [reply]
The issue is fixed and already part of the current [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous). Again thanks for the report.

**Reply by oldsparks** [reply]
Fast work Marko and thank you! I'm glad it was somewhat trivial. I'll try it out on the new laptop today. 

My sincere appreciation for your continued support. In another life I supported code for a government system that had to run 24x7x365. I was married to that system and chained at the ankles for 4 years of my life which is why I have no desire to code any longer.

**Reply by MAKOMO** [reply]
Thanks for your kind words! 

The issue is within the Qt framework Artisans UI is based on which handles also the storage of the app settings. For this each setting has somehow be stored in a platform dependent ways. That works good for simple data types, but Artisan is a complex meshup of many libs. We use in parts a numeric library (partly implemented in C and Fortran) for computations for efficiency and numeric reasons which stores floats in a different way than within Artisans main implementation language Python. Here such "special" numbers leaked into settings in some cases and on some platforms (Windows) in some cases (not always), Qt had difficulties to read what it stored before. I did ensure now that those number no longer leak into the settings and ensured that issues on reading those "broken" settings will succeed.

I understand your "marriage" as I feel the same with Artisan.
